lib/sober_swag/reporting/output/interface.rb

```diff
@@ -0,0 +1,77 @@
+module SoberSwag
+  module Reporting
+    module Output
+      ##
+      # Interface methods for all outputs.
+      module Interface
+        def call!(item)
+          res = serialize_report(item)
+
+          raise Report::Error.new(res) if res.is_a?(Report::Base) # rubocop:disable Style/RaiseArgs
+
+          res
+        end
+
+        ##
+        # Show off that this is a reporting output.
+        def reporting?
+          true
+        end
+
+        ##
+        # Delegates to {#call}
+        def serialize(item)
+          call(item)
+        end
+
+        def via_map(&block)
+          raise ArgumentError, 'block argument required' unless block
+
+          ViaMap.new(self, block)
+        end
+
+        def referenced(name)
+          Referenced.new(self, name)
+        end
+
+        def list
+          List.new(self)
+        end
+
+        ##
+        # Partition this serializer into two potentials.
```

Comment:
nit: It's slightly confusing why this is necessary, or in what circumstance this partitioning would be useful.

Reply:
9028365